server/src/apub/inbox/activities/create.rs

```diff
@@ -65,7 +65,7 @@ async fn receive_create_post(
 
   let post = PostForm::from_apub(&page, client, pool, Some(user.actor_id()?)).await?;
 
-  let inserted_post = blocking(pool, move |conn| Post::create(conn, &post)).await??;
+  let inserted_post = blocking(pool, move |conn| upsert_post(&post, conn)).await??;
```

Comment:
Could you add a quick comment on why we are using upsert instead of create here?

Reply:
Done.